protobuf/jaeger.proto

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package jaegertracing.protobuf;
+
+message Empty {};
+
+message ServiceRequest {
+    string service_name = 1;
+}
+
+message Tag {
+    string key = 1;
+    oneof value {
+        string str_value = 2;
+        double double_value = 3;
+        bool bool_value = 4;
+        int64 long_value = 5;
+        bytes binary_value = 6;
+    }
+}
+
+message Log {
+    int64 timestamp = 1;
+    repeated Tag fields = 2;
+}
+
+message SpanRef {
+    enum Type {
+        CHILD_OF = 0;
+        FOLLOWS_FROM = 1;
+    }
+    Type type = 1;
+    uint64 trace_id_high = 2;
+    uint64 trace_id_low = 3;
```

Comment:
I think I would prefer these two fields to be encapsulated into a TradeId struct, however there is a question of efficiency in the generated Go types. For example, Thrift would define traceId field as a pointer in such case, which requires an additional allocation for no reason.

This is actually a more general question. Before committing the IDLs, we should really run perf tests to compare thrift/tchannel vs. proto/grpc, in terms of throughput, latency, memory, CPU.

Reply:
Interesting question. For sure agree about the benchmarking. Actually started this change after reading about the different encoding/decoding benchmarks where it seems Protobuf has the best performance in Go, and presumably elsewhere. To set the record straight, when it comes to Go, there is the "fast" generator [gogo/protobuf](https://github.com/gogo/protobuf) vs. the standard generator [golang/protobuf](https://github.com/golang/protobuf). The reputation for speed is only true for the fast generator that I use throughout.